### PR TITLE
Task 6

### PR DESCRIPTION
**Pull request by valakar**
## Evaluation criteria

Reviewers should verify the lambda functions, SQS and SNS topic and subscription in PR.

- [x]  **1** - File **serverless.yml** contains configuration for **catalogBatchProcess** function
- [x]  **2** - File **serverless.yml** contains policies to allow lambda **catalogBatchProcess** function to interact with SNS and SQS
- [x]  **3** - File **serverless.yml** contains configuration for SQS **catalogItemsQueue**
- [x]  **4** - File **serverless.yml** contains configuration for SNS Topic **createProductTopic** and email subscription

## Additional (optional) tasks

- [x]  **+1** - **catalogBatchProcess** lambda is covered by **unit** tests
- [x]  **+1** - set a Filter Policy for SNS **createProductTopic** in **serverless.yml** (Create an additional email subscription and distribute messages to different emails depending on the filter for any product attribute)

***
##### Deployed applications
[Cloudfront](https://d1jpnfaozgam1v.cloudfront.net/)

***
##### Documentation
[Swagger](https://app.swaggerhub.com/apis/valakar/NodeAWSBE/0.0.3) - all requests should work (was not updated)

### CODE REVIEW

File: product-service/serverless.ts [hunk 1]
```diff
@@ -1,72 +1,150 @@
...
+            SNSSubscriptionBackup: {
+                Type: 'AWS::SNS::Subscription',
+                Properties: {
+                    Endpoint: config.BACKUP_EMAIL,
+                    Protocol: 'email',
+                    TopicArn: {
+                        Ref: 'SNSTopic'
+                    },
+                    FilterPolicy: {
+                        "tier": ["Mighty"]
```
Comment (OHalahan) [inline product-service/serverless.ts]: am I missing something? It looks like this filter policy is never invoked as no parameters are passed while publishing SNS notification

File: product-service/src/services/sns.service.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+export class SNSService {
+    private readonly sns = new SNS({ region: config.REGION });
+    private readonly topicArn = config.SNS_TOPIC_ARN;
+
+    sendEmail(subject: string, body: string): Promise<PromiseResult<PublishResponse, AWSError>> {
+        const a = this.sns.publish({
+            Subject: subject,
+            Message: body,
+            TopicArn: this.topicArn
```
Comment (OHalahan) [inline product-service/src/services/sns.service.ts]: here `tier` is not exposed to the filter policy

File: product-service/serverless.ts (hunk 1 above)
Comment (valakar) [reply product-service/serverless.ts]: Hi Oleksandr, thank you for pointing to this issue. It was my mistake, I have uploaded fix

**Review by OHalahan — APPROVED** [review]

File: product-service/serverless.ts (hunk 1 above)
Comment (OHalahan) [reply product-service/serverless.ts]: Hi @valakar

Thank you for cooperation. Grade is updated.

**Pull request merged**